anagrams.py

```diff
@@ -35,11 +41,20 @@ def find_anagrams(words):
         {'dgo': ['dog'], 'act': ['cat', 'act']}
 
     """
-    anagrams = {
-        alphabetize(word): [
-            w for w in words
-            if alphabetize(w) == alphabetize(word)]
-        for word in words}
+    # Given solution that needed to be adjusted to perform better
+    # anagrams = {
+    #     alphabetize(word): [
+    #         w for w in words
+    #         if alphabetize(w) == alphabetize(word)]
+    #     for word in words}
+
+    anagrams = {}
+    for word in words:
+        if alphabetize(word) in anagrams:
+            anagrams[alphabetize(
+                word)].append(word)
+        else:
+            anagrams[alphabetize(word)] = [word]
```

Comment:
There is a lot of alphabetizing going on here .. any way to do it less often?  Each alphabetize costs a 'sort' operation.